File: CalculatorBackend/src/computer/ReplacePowersDecorator.java
package computer;

import core.Main;
import ifc.IfcExpressionProcessor;

public class ReplacePowersDecorator implements IfcExpressionProcessor {
    private final IfcExpressionProcessor next;
    
    public ReplacePowersDecorator(IfcExpressionProcessor next) {
        this.next = next;
    }

    @Override
    public String process(String expression) {
        expression = Main.replacePowers(expression);
        return next.process(expression);
    }
}

File: computer/ReplacePowersDecorator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using CalculatorBackend.ifc;
using CalculatorBackend;

internal class ReplacePowersDecorator : IfcExpressionProcessor
{
    private readonly IfcExpressionProcessor _next;

    public ReplacePowersDecorator(IfcExpressionProcessor next)
    {
        _next = next;
    }

    public string Process(string expression)
    {
        expression = ACalculator.ReplacePowers(expression);
        return _next.Process(expression);
    }
}
